lib/vm/src/context.rs

```diff
@@ -11,12 +12,18 @@ use cosmwasm::traits::{Api, Storage};
 use crate::memory::{read_region, write_region};
 use cosmwasm::types::{CanonicalAddr, HumanAddr};
 
+/// An undocumented, unstable constant. This can change at any time. Be warned.
+static WRITE_REGION_ERROR: i32 = -1001;
+
 pub fn do_read<T: Storage>(ctx: &mut Ctx, key_ptr: u32, val_ptr: u32) -> i32 {
     let key = read_region(ctx, key_ptr);
     let mut value: Option<Vec<u8>> = None;
     with_storage_from_context(ctx, |store: &mut T| value = store.get(&key));
     match value {
-        Some(buf) => write_region(ctx, val_ptr, &buf),
+        Some(buf) => match write_region(ctx, val_ptr, &buf) {
+            Ok(bytes_written) => bytes_written.try_into().unwrap(),
+            Err(_) => WRITE_REGION_ERROR,
```

Comment:
Okay, this is the point I mentioned before. That is now saying "too small, try with 1001 bytes". I realize the retrun value of c_read is not documented (bad me). However, thiis is the relevant code, and a good fix while working on this memory stuff: https://github.com/confio/cosmwasm/blob/master/src/imports.rs#L54-L60